### PR TITLE
✨ Integrate Motion screens

### PR DESCRIPTION
**Pull request by warrenseine**
# Problem

The Motion flow needs custom screens to be integrated within the router.

# Solution

Add the new screens. Some are incomplete (e.g. faking upload), because they will require further business logic integration. This change is mostly about the UI.

Everything is behind a feature flag.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [ ] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: src/components/ActiveVideo/ActiveVideo.tsx [hunk 1]
```diff
@@ -4,52 +4,38 @@ import { useState } from 'preact/hooks'
...
+      <FaceNotDetected restart={() => setError(null)} translate={translate} />
+    )
+  } else if (error) {
+    console.error(`Unsupported error: ${error}`)
   }
 
   // See: https://github.com/preactjs/preact/issues/2748
   return (
     <ActiveVideoCapture
       debug={true}
```
Comment (warrenseine) [inline src/components/ActiveVideo/ActiveVideo.tsx:72]: Useful for now. Will be removed when reaching alpha status.

File: src/components/ActiveVideo/Uploading/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const Uploading: FunctionComponent<Props> = ({ nextStep }) => {
+  const [uploadProgress, setUploadProgress] = useState(0)
+
+  let interval: NodeJS.Timeout
+
+  /**
+   * This method is only intended to mimic the real upload.
+   */
+  useEffect(() => {
+    interval = setInterval(() => {
```
Comment (warrenseine) [inline src/components/ActiveVideo/Uploading/index.tsx:34]: This is a temporary fake upload while we work on integrating Biometrics Media Uploader.

**Review by osolliec — COMMENTED** [review]
LGTM just a few comments to understand a bit more the context of the different @onfido/active-video-capture package

File: src/components/ActiveVideo/BaseScreen/index.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { h, FunctionComponent, ComponentChildren } from 'preact'
```
[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/ActiveVideo/BaseScreen/index.tsx:1]: They are only used within a [separate internal demo app](https://deploy-previews.eu-west-1.dev.onfido.xyz/onfido/biometrics/experience/active-video-capture/master/index.html), but this is almost 100% copy-pasted.

While there is no plan per-se, I expect that we'll remove clean up the whole integration once we're out of alpha status.

File: package.json [hunk 4]
```diff
@@ -244,7 +244,7 @@
     "yn": "^3.0.0"
   },
   "dependencies": {
-    "@onfido/active-video-capture": "^0.0.1",
+    "@onfido/active-video-capture": "^0.5.0",
```
[the remark this answers is not in the document]
Comment (warrenseine) [reply package.json:247]: The package exposes the main capture experience. While it's not a goal in itself, we want to ensure we're free to distribute the capture experience outside of the Web SDK.

File: src/components/Router/StepComponentMap.tsx [hunk 5]
```diff
@@ -248,17 +269,44 @@ const buildFaceComponents = (
...
+  const allActiveVideoSteps = [
+    ActiveVideoIntro,
+    LazyActiveVideo,
+    ActiveVideoRecordingComplete,
+    ActiveVideoUpload,
+  ]
+
+  if (mobileFlow) {
+    // do not display intro on cross device flow
+    // @ts-ignore
```
[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/Router/StepComponentMap.tsx:308]: I can certainly have a look. I don't like it either.

File: src/components/ActiveVideo/Uploading/index.tsx [hunk 6]
```diff
@@ -6,15 +6,25 @@ import { Wrapper } from '../Wrapper'
...
-const Uploading: FunctionComponent<Props> = ({ nextStep }) => {
+const mapStateToProps = ({ captures }: RootState, props: Props): Props => ({
+  ...props,
+  videoPayload: captures[buildCaptureStateKey({ method: 'activeVideo' })]?.blob,
+})
+
+const Uploading: FunctionComponent<Props> = ({ nextStep, videoPayload }) => {
   const [uploadProgress, setUploadProgress] = useState(0)
 
+  console.log(videoPayload)
```
[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/ActiveVideo/Uploading/index.tsx:26]: I will add a separate commit where I implement the logic of this screen (the actual upload) and remove it.

File: src/components/Router/StepComponentMap.tsx [hunk 7]
```diff
@@ -289,9 +289,16 @@ const buildActiveVideoComponents = (
   mobileFlow?: boolean,
   isFirstCaptureStepInFlow?: boolean
 ): ComponentType<StepComponentProps>[] => {
+  // Active Video Capture SDK already includes a footer and should appear under a
+  // semi-transparent navigation bar. There is no way to set it at the step level
+  // because it only impacts the capture screen. As a (horrible) workaround,
+  // `StepsRouter` will look at the component name to set `edgeToEdgeContent`.
+  const ActiveVideoCapture = LazyActiveVideo
+  ActiveVideoCapture.displayName = 'ActiveVideoCapture'
```
[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/Router/StepComponentMap.tsx:297]: That's an excellent idea. I'm on it.

This is a horrible hack, and I hope to find a better way once we've managed to release it for the next code cut-off 🙏🏻

File: src/components/ActiveVideo/Uploading/index.tsx [hunk 8]
```diff
@@ -58,4 +68,7 @@ const Uploading: FunctionComponent<Props> = ({ nextStep }) => {
   )
 }
 
-export default localised(Uploading)
+export default connect(mapStateToProps)(
+  // @ts-ignore: Preact Provider compatibility
```
[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/ActiveVideo/Uploading/index.tsx:72]: I've noticed many components have this @ts-ignore comment. I don't think it's easy to fix.

File: src/components/Router/StepComponentMap.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/Router/StepComponentMap.tsx]: I've removed all the @ts-ignore and TODOs in this file! I'm still relying on `unknown` just for the Preact / Redux type compatibility. It's isolated so I think it's fine. Have a look at this [commit](https://github.com/onfido/onfido-sdk-ui/pull/1965/commits/28e4e67b1cfbf1dc25a4b2f20106251e682f9052).

File: src/components/ActiveVideo/Uploading/index.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (osolliec) [reply src/components/ActiveVideo/Uploading/index.tsx]: yeah I didn't mean anything by this comment other than Preact can be a pain, apparently

File: src/components/Router/StepComponentMap.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (osolliec) [reply src/components/Router/StepComponentMap.tsx:297]: thank you. We use both react testing library and enzyme (deprecated), please write it using RTL :)

[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/Router/StepComponentMap.tsx:297]: I've implemented [here](https://github.com/onfido/onfido-sdk-ui/pull/1965/commits/7e36f73f8ca9bc005cb60eac87a5728560f26229). It's not pretty as I'm checking for the `transparent` class name and relying on the specific order of components in the step, but it's better than nothing. Again, we'll try to remove this workaround later.

File: src/components/ActiveVideo/Uploading/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (warrenseine) [reply src/components/ActiveVideo/Uploading/index.tsx]: It's implemented in [this commit](https://github.com/onfido/onfido-sdk-ui/pull/1965/commits/fc01f98d9cc2df99f18fffdac478d4a3ee257a8d).

**Review by osolliec — APPROVED** [review]
LGTM, ideally a 2nd reviewer could be nice as I'm still a newbie in the team

**Pull request merged**